tests/ignite/metrics/test_ndcg.py

```diff
@@ -8,62 +8,78 @@
 from ignite.metrics.recsys.ndcg import NDCG
 
 
-@pytest.mark.parametrize(
-    "y_pred, y_true",
-    [
-        (torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])),
-        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]])),
+@pytest.fixture(params=[item for item in range(5)])
+def test_case(request):
+
+    return [
+        (torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]]), True),
+        (torch.tensor([[3.7, 4.8, 3.9, 4.3, 4.9]]), torch.tensor([[2.9, 5.6, 3.8, 7.9, 6.2]]), True),
         (
             torch.tensor([[0.1, 0.2, 0.3, 0.4, 0.5], [3.7, 4.8, 3.9, 4.3, 4.9], [3.7, 4.8, 3.9, 4.3, 4.9]]),
             torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.2, 4.5, 8.9, 5.6, 7.2], [2.9, 5.6, 3.8, 7.9, 6.2]]),
+            True,
         ),
-        (torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]])),
+        (torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7]]), torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0]]), False),
         (
             torch.tensor([[3.7, 3.7, 3.7, 3.7, 3.7], [3.7, 3.7, 3.7, 3.7, 3.9]]),
             torch.tensor([[1.0, 2.0, 3.0, 4.0, 5.0], [1.0, 2.0, 3.0, 4.0, 5.0]]),
+            False,
         ),
-    ],
-)
+    ][request.param]
+
+
 @pytest.mark.parametrize("k", [None, 2, 3])
-def test_output_cpu(y_pred, y_true, k):
+@pytest.mark.parametrize("exponential", [True, False])
+def test_output_cpu(test_case, k, exponential):
 
     device = "cpu"
-
-    ndcg = NDCG(k=k, device=device)
+    y_pred, y_true, try_ignore_ties = test_case
+    ndcg = NDCG(k=k, device=device, exponential=exponential, ignore_ties=False)
     ndcg.update([y_pred, y_true])
     result_ignite = ndcg.compute()
-    result_sklearn = ndcg_score(y_true.numpy(), y_pred.numpy(), k=k)
+
+    if try_ignore_ties:
+        ndcg = NDCG(k=k, device=device, exponential=exponential, ignore_ties=True)
+        ndcg.update([y_pred, y_true])
+        result_ignite_ignore_ties = ndcg.compute()
```

Comment:
Why can't we code it as 
```python
ndcg = NDCG(k=k, device=device, exponential=exponential, ignore_ties=try_ignore_ties)
ndcg.update([y_pred, y_true])
result_ignite = ndcg.compute()

result_sklearn = ndcg_score(y_true.numpy(), y_pred.numpy(), k=k, ignore_ties=try_ignore_ties)
```
?

Reply:
We can. But we will end up computing the `ignore_ties = False` twice in case of `try_ignore_ties = True` which is inefficient and in my personal opinion a little confusing. However if you say so, I can do it this way